app/src/main/java/org/simple/clinic/registration/facility/RegistrationFacilitySelectionScreenController.kt

```diff
@@ -213,11 +214,14 @@ class RegistrationFacilitySelectionScreenController @Inject constructor(
     return events
         .ofType<RegistrationFacilityConfirmed>()
         .map { it.facilityUuid }
-        .flatMap { facilityUuid ->
-          userSession.ongoingRegistrationEntry()
-              .map { it.copy(facilityId = facilityUuid) }
-              .doOnSuccess(userSession::saveOngoingRegistrationEntry)
-              .flatMapCompletable { userSession.saveOngoingRegistrationEntryAsUser() }
+        .map { facilityUuid ->
+          val entry = (userSession.ongoingRegistrationEntry() as Just).value
+          entry.copy(facilityId = facilityUuid)
```

Comment:
Should we provide a method to change this from `OngoingRegistrationEntry` instead of using `copy` directly? WDYT?

Reply:
It's a good idea, I'll do this when I migrate `RegistrationFacilitySelectionScreenController` to Mobius.